lib/ansible/plugins/loader.py

```diff
@@ -1005,14 +1005,19 @@ def all(self, *args, **kwargs):
 
         loaded_modules = set()
         for path in all_matches:
-            name = os.path.splitext(path)[0]
-            basename = os.path.basename(name)
+            if isinstance(self, Jinja2Loader):
+                # file are names not unique, only filter/test names are
+                # so we cache path in this case for filter/test files
+                name = basename = path
```

Comment:
While this patch solves the issue I'm seeing, setting `basename = path` seems wrong to me, I think it just masks the underlying logic error? So maybe a more thorough fix in the caching code would be needed?

Reply:
tru